Src/AutoFixture/Kernel/FactoryMethodQuery.cs

```diff
@@ -39,6 +43,7 @@ public IEnumerable<IMethod> SelectMethods(Type type)
             return from mi in type.GetMethods(BindingFlags.Static | BindingFlags.Public)
                    where mi.ReturnType == type
                    let parameters = mi.GetParameters()
+                   where mi.GetParameters().All(p => p.ParameterType != type)

```

Comment:
That's interesting. The intention was to **simplify** a specimen creation. Please let me describe it in details.

I needed to create an instance of the `System.IO.Stream` class. The class is abstract but I did have AutoNSubstitute on board so that should not be a problem. Instead I received the circular reference exception and started investigating the code.

It took a while to skim across ~900 lines of decompiled code to find out a single 'pseudo' factory method that causes the problem (and blocks AutoNSubstitute BTW):

```csharp
public static Stream Synchronized(Stream stream)
```

So, my intention was to exclude such kind of methods from the specimen creation pipeline at all. If the inability to create an instance is *easily* detectable, why should one need deal with the circular reference errors? ;)

In advance it allows to instantiate the following type:

```csharp
public class TypeWithRealAndPseudoFactoryMethod
{
    private TypeWithRealAndPseudoFactoryMethod() { }

    // Not a real factory method. Causes circular refs but
    // will be taken by default as the most modest one
    public static TypeWithRealAndPseudoFactoryMethod DoSomething(
        TypeWithRealAndPseudoFactoryMethod argument)
    {
        return new TypeWithRealAndPseudoFactoryMethod();
    }

    // valid factory method, but it will be ignored by AF since 
    // it's looking for the most modest method first
    public static TypeWithRealAndPseudoFactoryMethod Create(object arg1, object arg2)
    {
        return new TypeWithRealAndPseudoFactoryMethod();
    }
}
```

Does it make more sense?